src/server/scripts/EasternKingdoms/Karazhan/boss_prince_malchezaar.cpp

```diff
@@ -174,10 +151,60 @@ struct boss_malchezaar : public BossAI
         instance->HandleGameObject(instance->GetGuidData(DATA_GO_NETHER_DOOR), true);
     }
 
+    bool CheckForbidden(uint8 choice, std::list<uint8> toCheckList)
+    {
+        for (uint8 check : toCheckList)
+        {
+            if (check == choice)
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    void SpawnInfernal(Creature* relay, Creature* target)
+    {
+        if (Creature* infernal = relay->SummonCreature(NPC_NETHERSPITE_INFERNAL, target->GetPosition(), TEMPSUMMON_TIMED_DESPAWN, 180000))
+        {
+            infernal->SetDisplayId(INFERNAL_MODEL_INVISIBLE);
+            relay->CastSpell(target, SPELL_INFERNAL_RELAY_TWO);
+            relay->CastSpell(infernal, SPELL_INFERNAL_RELAY);
+            infernal->SetFaction(me->GetFaction());
+            infernal->SetControlled(true, UNIT_STATE_ROOT);
+            relay->CastSpell(infernal, SPELL_INFERNAL_RELAY);
+            summons.Summon(infernal);
+        }
+    }
+
+    bool MaxSpawns(std::list<Creature*> spawns)
+    {
+        return spawns.size() == 0;
+    }
+
+    Creature* PickTarget(std::list<Creature*> pickList)
+    {
+        uint8 index = urand(0, pickList.size()-1);
+        uint8 counter = 0;
```

Comment:
You might be able to do less passes with something like:
`std::list<Creature*>::iterator it = availableTargets.begin();
std::advance(it, 2);
	
Creature* result = *it;
availableTargets.erase(it);
	
return result;`

You can then remove the ejection of the selected target from the outside function, having this one deal with all the management of selecting your target

Reply:
Obviously returning null should there be no entry to pick from